File: frontend/src/types/photo.ts
export type PhotoStatus = 'UPLOAD_REQUESTED' | 'RECEIVED' | 'AVAILABLE' | 'HIDDEN' | 'REMOVED';

export interface Photo {
  id: string;
  originalFilename: string | null;
  objectKey: string | null;
  contentType: string | null;
  sizeBytes: number | null;
  status: PhotoStatus;
  favorite: boolean;
  guestName: string | null;
  guestMessage: string | null;
  createdAt: string;
  downloadUrl: string | null;
}

export interface GuestUploadItemResponse {
  photoId: string;
  objectKey: string | null;
  status: PhotoStatus;
  originalFilename: string | null;
}

export interface GuestUploadResponse {
  uploadedCount: number;
  photos: GuestUploadItemResponse[];
  message: string;
}

export interface PhotoFavoriteUpdateRequest {
  favorite: boolean;
}

export interface PhotoStatusUpdateRequest {
  status: PhotoStatus;
}

File: backend/src/main/java/com/memora/core/domain/model/Photo.java
package com.memora.core.domain.model;

import java.time.LocalDateTime;
import java.util.UUID;
import lombok.Builder;
import lombok.Value;

@Value
@Builder(toBuilder = true)
public class Photo {
	UUID id;
	UUID eventId;
	String originalFilename;
	String objectKey;
	String contentType;
	Long sizeBytes;
	PhotoStatus status;
	boolean favorite;
	int likesCount;
	String guestName;
	String guestMessage;
	UUID uploadGroupId;
	LocalDateTime createdAt;
	LocalDateTime updatedAt;
}
